scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb

```diff
@@ -400,10 +352,37 @@ namespace <%= namespace %>.SDK {
             return Delete(variantId);
         }
 
-        private void OnCartLineItemChange(CartLineItem modifiedLineItem) {
-            if (!LineItemsNew.Contains(modifiedLineItem) && !LineItemsModified.Contains(modifiedLineItem)) {
-                LineItemsModified.Add(modifiedLineItem);
+        public void UpdateLineItemsFromCheckoutLineItems(List<CheckoutLineItem> checkoutLineItems) {
+            foreach(CheckoutLineItem checkoutLineItem in checkoutLineItems) {
+                bool didUpdateID = false;
+
+                foreach(CartLineItem cartLineItem in LineItems) {    
+                    if (cartLineItem.VariantId == checkoutLineItem.variant().id()) {
```

Comment:
Spoke with @mikkoh, Voiced my concerns about the runtime of the nested for loops